### ISSUE TITLE
bug: SELECT * FROM BUG_12595 WHERE a LIKE 'ha|%an' ESCAPE '|';The query result is empty 

### ISSUE THREAD

**Issue by shangyanwen** [stoneatom/stonedb#272]
### Describe the problem

```
SET @OLD_SQL_MODE12595=@@SQL_MODE, @@SQL_MODE='';
SHOW LOCAL VARIABLES LIKE 'SQL_MODE';

CREATE TABLE BUG_12595(a varchar(100))engine=stonedb;
INSERT INTO BUG_12595 VALUES ('hakan%'), ('hakank'), ("ha%an");

SELECT * FROM BUG_12595 WHERE a LIKE 'ha|%an' ESCAPE '|';
##test result
Empty set (0.02 sec)
```

### Expected behavior

```
##innodb test result
mysql> SELECT * FROM BUG_12595 WHERE a LIKE 'ha|%an' ESCAPE '|';
+-------+
| a     |
+-------+
| ha%an |
+-------+
1 row in set (0.07 sec)
```

### How To Reproduce

```
SET @OLD_SQL_MODE12595=@@SQL_MODE, @@SQL_MODE='';
SHOW LOCAL VARIABLES LIKE 'SQL_MODE';

CREATE TABLE BUG_12595(a varchar(100))engine=stonedb;
INSERT INTO BUG_12595 VALUES ('hakan%'), ('hakank'), ("ha%an");

SELECT * FROM BUG_12595 WHERE a LIKE 'ha|%an' ESCAPE '|';
```

### Environment

1. StoneDB for mysql5.7 (release)
2. Ubuntu 20.04.4

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

**Comment by adofsauron** [comment]
ACK

**Comment by adofsauron** [comment]
the same as https://github.com/stoneatom/stonedb/issues/271

**Comment by RingsC** [comment]
fixed in PR #1002 . and close it manually.
